src/compiler/typescript_to_rust/azle_generate/src/guard_function/rust.rs

```diff
@@ -4,15 +4,16 @@ use quote::quote;
 pub fn generate(function_name: &String) -> TokenStream {
     let function_call = format!("{function_name}()");
     quote! {
-        BOA_CONTEXT_REF_CELL.with(|box_context_ref_cell| {
-            let mut boa_context = box_context_ref_cell.borrow_mut();
+        BOA_CONTEXT_REF_CELL.with(|boa_context_ref_cell| {
+            let mut boa_context = boa_context_ref_cell.borrow_mut();
 
-            let js_guard_fn_return_value =
-                unwrap_boa_result(boa_context.eval_script(boa_engine::Source::from_bytes(#function_call)), &mut boa_context);
+            let js_guard_fn_return_value = boa_context
+                .eval_script(boa_engine::Source::from_bytes(#function_call))
+                .unwrap_or_trap(&mut boa_context);
 
             match js_guard_fn_return_value.try_from_vm_value(&mut *boa_context) {
                 Ok(return_value) => return_value,
-                Err(e) => ic_cdk::api::trap(&format!("Uncaught TypeError: {}",&e.0))
+                Err(e) => ic_cdk::api::trap(&format!("Uncaught {}", &e.0)),
```

Comment:
Guard functions return a `Result<(), String>`. Can we discuss if the `unwrap_or_trap` is actually necessary? I've updated the function so that it doesn't call trap, it just returns an `Err(string)` instead which I think is sufficient.